### PR TITLE
Addition of a  useRefreshTime hook

### LINKED ISSUE #121: Refactor logics for refresh time intervals to own hook useRefreshTime(...)
We have refresh time logics in several screens, which are handled by an `useEffect` each in the same way with different params. There could be a new hook handling this generally.

### PR DESCRIPTION
**Pull request by digorath**
## Changes proposed in this PR:

As described in #121 there was a lot of code duplication for the refreshTime.
It is now centralised in a new custom hook.

There should be no visual changes.

The default value for the second argument was chosen because there was only one place that differed from it.

---

Fixes #121

### CODE REVIEW

File: src/hooks/RefreshTime.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { useCallback, useEffect, useState } from 'react';
+
+import { consts } from '../config';
+import { refreshTimeFor } from '../helpers';
+
+export const useRefreshTime = (
+  refreshTimeKey: string,
+  refreshInterval = consts.REFRESH_INTERVALS.STATIC_CONTENT
```
Comment (donni106) [inline src/hooks/RefreshTime.ts:8]: maybe we should export `const ONCE_A_DAY = 'ONCE_A_DAY';` in `consts` to use it here, as `STATIC_CONTENT` is logically used for data of type static contents. the value is the same, but the meaning is a different.

**Comment by donni106** [comment]
I plan to merge this for 1.4.2 release next week.

File: src/hooks/RefreshTime.ts (hunk 1 above)
Comment (donni106) [reply src/hooks/RefreshTime.ts]: done in https://github.com/ikuseiGmbH/smart-village-app-app/pull/142/commits/b525e8503899998113202a45cec465cccab2b50a

**Review by donni106 — APPROVED** [review]

**Comment by SoleCincis** [comment]
Why useRefreshTime? 
We don't want to copy and paste the same logic within all the components. So what we do instead, is to extract the logic. And move it to its own function. So any component that may need the logic, can import it in, and use it.

**Comment by SoleCincis** [comment]
So refreshTime and RefreshControl both refetch data , just first one is doing it with an interval of once per day and other one is doing it on user action ( pull screen down ) ? 
If yes could we use this new custom hook somehow in RefreshControl as well ?

**Review by SoleCincis — COMMENTED** [review]
doubts have been clarified.

**Pull request merged**